x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Count.java

```diff
@@ -87,4 +93,30 @@ public Nullability nullable() {
     protected TypeResolution resolveType() {
         return EsqlTypeResolutions.isExact(field(), sourceText(), DEFAULT);
     }
+
+    @Override
+    public Expression surrogate() {
+        var s = source();
+        var field = field();
+
+        if (field.foldable()) {
+            if (field instanceof Literal l) {
+                if (l.value() != null && (l.value() instanceof List<?>) == false) {
+                    // TODO: Normalize COUNT(*), COUNT(), COUNT("foobar"), COUNT(1) as COUNT(*).
+                    // Does not apply to COUNT([1,2,3])
+                    // return new Count(s, new Literal(s, StringUtils.WILDCARD, DataTypes.KEYWORD));
+                    return null;
+                }
+            }
+
+            // COUNT(const) is equivalent to MV_COUNT(const)*COUNT(*) if const is not null; otherwise COUNT(const) == 0.
+            return new Mul(
+                s,
+                new Coalesce(s, new MvCount(s, field), List.of(new Literal(s, 0, DataTypes.INTEGER))),
+                new Count(s, new Literal(s, StringUtils.WILDCARD, DataTypes.KEYWORD))
```

Comment:
Elsewhere in the code we use the convention of Count(1) - let's apply that here as well.

Reply:
Hm, I tried to look for other places where `Count(1)` is used, but couldn't verify this (for ESQL at least).

Parsing `count()` results in `new Count(source, "*", KEYWORD)`, we also use the same in `Sum.surrogate()` and the (currently disabled) `NormalizeAggregate` used to replace all `count(literal)` by `count(*)` as well.

For consistency, I'd like to stick to `count(*)` here.